File: p11.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int maxArea(vector<int>& height) {
        int lo, hi;
        lo = 0, hi = height.size() - 1;
        int ret = 0;
        while(lo < hi){
            ret = max(ret, (hi - lo) * min(height[lo], height[hi]));
            if(height[lo] > height[hi]) hi--;
            else lo++;
        }
        return ret;
    }
};

File: p11.java
public class p11 {
    public int maxArea(int[] height) {
        int lo = 0;
        int hi = height.length - 1;
        int mx = -1;

        while(lo < hi){
            int here = (hi - lo) * Math.min(height[lo], height[hi]);
            mx = Math.max(mx, here);

            if(height[lo] > height[hi]) hi--;
            else lo++;
        }
        
        return mx;
    }
}
